src/main/webapp/script.js

```diff
@@ -12,7 +12,118 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/**
+ * Submits a fetch request to the /data url. Upon receiving a JSON encoding of the
+ * nodes and edges of the graph, renders the graph in a container on the page using
+ * the cytoscape.js library
+ */
+async function generateGraph() {
+  // Arrays to store the cytoscape graph node and edge objects
+  let graphNodes = [];
+  let graphEdges = [];
+  const response = await fetch("/data");
 
-function testFunc() {
-  fetch("/data").then(response => true);
-}
\ No newline at end of file
+  const serverErrorStatus = response.headers.get("serverError");
+
+  // Error on server side
+  if (serverErrorStatus !== null) {
+    displayError(serverErrorStatus);
```

Comment:
This displays the error but does not terminate function execution. Missing a return statement?